rocm-cmake version cannot be used by cmake

The version specified in rocm-cmake is not written to a file detectable by cmake, so other packages cannot require a specific version of rocm-cmake. Attempting to require version 0.5 (the latest version on master) results in the error message:
```CMake Error at cmake/Dependencies.cmake:119 (find_package):
[cmake]   Could not find a configuration file for package "ROCM" that is compatible
[cmake]   with requested version "0.5".
[cmake] 
[cmake]   The following configuration files were considered but not accepted:
[cmake] 
[cmake]     /opt/rocm/share/rocm/cmake/ROCMConfig.cmake, version: unknown
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,10 @@
 # Change Log for rocm-cmake
 
 
+## [0.5.1]
+### Added
+    - Add ROCMConfigVersion.cmake file so cmake can check the version
+
 ## [0.5]
 ### Added
     - Change Log added and version number incremented

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@
 ## [0.5.1]
 ### Added
     - Add ROCMConfigVersion.cmake file so cmake can check the version
+    - Switched to semantic versioning
 
 ## [0.5]
 ### Added

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,12 +9,17 @@ set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE PATH "")
 project(rocm-cmake)
 
 install(DIRECTORY share DESTINATION .)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/ROCMConfigVersion.cmake DESTINATION share/rocm/cmake)
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/share/rocm/cmake)
 include(ROCMCreatePackage)
 include(ROCMSetupVersion)
 
 rocm_setup_version(VERSION 0.5)
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/ROCMConfigVersion.cmake
+    COMPATIBILITY SameMinorVersion)
 
 rocm_create_package(
     NAME rocm-cmake

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@ list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/share/rocm/cmake)
 include(ROCMCreatePackage)
 include(ROCMSetupVersion)
 
-rocm_setup_version(VERSION 0.5)
+rocm_setup_version(VERSION 0.5.1)
 
 include(CMakePackageConfigHelpers)
 write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/ROCMConfigVersion.cmake

